src/clj/game/cards/hardware.clj

```diff
@@ -653,9 +653,9 @@
     :leave-play (req (remove-watch state :obelus)
                      (lose state :runner :hand-size {:mod (:tag runner)}))
     :events {:successful-run-ends {:once :per-turn
-                                   :req (req (let [successes (rest (turn-events state side :successful-run))]
+                                   :req (req (let [successes (turn-events state side :successful-run-ends)]
                                                (and (#{[:rd] [:hq]} (:server target))
-                                                    (empty? (filter #(#{'(:rd) '(:hq)} %) successes)))))
+                                                    (empty? (filter #(some #{:rd :hq} (:server (first %))) successes)))))
```

Comment:
I'm sure this check if empty after filter has a more compact function is checking no entry fultils the predicate.